src/main/java/com/swirlds/recordserver/AccountsService.java

```diff
@@ -47,10 +52,84 @@ public AccountsService(Config config) {
 
     @Override
     public void update(Routing.Rules rules) {
+        rules.get("/", this::getAccountsListMessageHandler);
         rules.get("/{idOrAliasOrEvmAddress}/nfts", this::getAccountNftsMessageHandler);
         rules.get("/{idOrAliasOrEvmAddress}", this::getAccountsMessageHandler);
     }
 
+    private void getAccountsListMessageHandler(ServerRequest request, ServerResponse response) {
+        final Optional<String> accountBalanceParam = request.queryParams().first("account.balance");
+        final Optional<String> accountIdParam = request.queryParams().first("account.id");
+        final Optional<String> publicKeyParam = request.queryParams().first("account.publickey");
+        final Optional<String> balanceParam = request.queryParams().first("balance");
+        final Optional<String> limitParam = request.queryParams().first("limit");
+        final Optional<String> orderParam = request.queryParams().first("order");
+        boolean includeBalance = false;
+        if(balanceParam.isPresent()) {
+            includeBalance = QueryParamUtil.parseQueryString(QueryParamUtil.Type._string, "token_id", balanceParam.get()).equals("true");
+        }
+        final int limit = parseLimitQueryString(limitParam);
+
+        JsonObjectBuilder accountResponseBuilder;
+        java.util.HashMap<Long, BalanceResponse> balanceMap = new java.util.HashMap<>();
+        //TODO: BalanceParam
+        if(!accountIdParam.isPresent()) {
+            final List<QueryParamUtil.WhereClause> whereClauses = new ArrayList<>();
+
+            //Add this for balances table
+            final QueryParamUtil.WhereClause accountBalanceWhereClause = accountBalanceParam.isPresent() ?
+                    QueryParamUtil.parseQueryString(QueryParamUtil.Type._long, "balance", accountBalanceParam.get()) : null;
+            java.util.Set<Long> ids = new java.util.HashSet<>();
+            String inEntityNumber = "";
+            if(accountBalanceWhereClause != null) {
+                getAccountBalanceRangeFilter(accountBalanceParam, whereClauses, limit, accountBalanceWhereClause);
+                whereClauses.add(new WhereClause(QueryParamUtil.Type._long, "token_id", Comparator.eq, "0"));
+                balanceMap = getBalanceAccountIds(whereClauses);
+                ids.addAll(balanceMap.keySet());
+                inEntityNumber = " and entity_number IN "+ ids ;
+            }
+
+            whereClauses.clear();
```

Comment:
this line seems to be an error -- are you sure that you want to clear all the where clauses already added?   (I might be wrong.)

Reply:
This is intentional because this endpoints queries multiple tables.
First the balance table, then I clear the where clauses to query the entity table, which the given ids.